### DISCUSSION TITLE
Private folders still shown in Albums, search, etc

### DISCUSSION THREAD

**Discussion by Ayo91** [photoprism/photoprism#4625] [Q&A]
I set a folder as private (similar to single images) and I still see images or videos related to that folder in the rest of the sections.

For search the checkbox for exclude private elements is enabled.

**Comment by graciousgrey** [comment]
Thank you for your feedback! 

This works as designed. Marking a folder as private in the folder edit dialog (which is only possible with "experimental features" enabled) only hides the folder in the  folder view. It does not mark all photos of this folder as private. 

You can mark photos as private as described here: 
- https://docs.photoprism.app/user-guide/organize/private/

The possibility to mark all photos of an album or folder as private is planned but not yet implemented. 
- https://github.com/photoprism/photoprism/issues/480

**Reply by Ayo91** [reply]
ok, now I understand the current implementation

**Comment by haecks** [comment]
Still a bit confused ...
I have just enabled `PhotoPrism® Plus Build 260418-c37fdf51a` i was very much hoping for getting back to a far early state, where private tagged albums weren't shown to every user. The pictures in that album are all tagged as private as well and the Admin-Option for hiding private content in global view is activated, but still even users granted only a viewer role are displayed such albums as well as their content. 
I am only sharing the Photoprism-Content with one other user locally, for whom i like to provide the pictures and albums not yet tagged as private, why i was hoping for getting back to that state of former functionality with buying into the Plus-Abo ...
At minimum, i was instantly hoping for restricting access to private content at least for users with limited `viewer`-role.
Seems like i am left with hoping for further development on the Multi-User topics.

**Reply by graciousgrey** [reply]
Photos that are marked as private https://docs.photoprism.app/user-guide/organize/private/ are not visible to users with role viewer or guest.

With experimental mode enabled you can also mark albums as private on the album edit dialog. However please note that this will only hide these albums from the album overview for users with the role viewer, it will not mark all photos of this album as private. So users with the viewer role may still have access to these photos if they have not been marked as private.

**Reply by haecks** [reply]
[UPDATE] User was not set to restricted `viewer`-role here.

<img width="940" height="764" alt="image" src="https://github.com/user-attachments/assets/737e07e2-4f1e-4303-bf74-b8d5f1bef4dc" />
Thank you for your answer! So this then remains an issue to my installation, i suppose. As shown, the "Viewer"-user with his `viewer`-role is still shown the `private`-marked album. Besides he still can enter that album and view the images tagged therein as `private` as well. Only the Lock-icon is shown indicating the `private`-status of the album and the pictures.
Following your description, that behaviour should not be correct. At least not, with the Admin-Option for hiding `private`-content globally activated, as it is set now.

I might try out with a fresh `viewer`-User to check.

**Reply by graciousgrey** [reply]
I cannot reproduce this. Neither on the latest nor on the preview docker image. I would suggest to confirm that the signed in user has role viewer, you can do so on the admin web ui: https://docs.photoprism.app/user-guide/users/

**Comment by haecks** [comment]
My bad! Very sorry about this!!! I was confident, due to the second line shown below the users-name, that that user was still on it's initial `viewer`-role ...

He is not!
Reset to that state, you are absolutely correct about that behaviour and he no longer sees the album nor it's images.

Currently there is no way to restrict other users from peeking into `private`-content, right?!?
That was the only thing i was hoping for, so that a regular user is able to help out archiving / sorting pictures in shared albums without bothering him with irrelevant content marked as `private` beforehand.

Thank you very much again for your immediate support!!!

**Reply by graciousgrey** [reply]
Glad to hear it works! Currently users with role guest and viewer cannot see privat content.
